include/drivers/arm/pl011.h

```diff
@@ -79,4 +81,20 @@
 
 #endif /* !PL011_GENERIC_UART */
 
+#define CONSOLE_T_PL011_BASE	CONSOLE_T_DRVDATA
+
+#ifndef __ASSEMBLY__
+
+#include <types.h>
+
+typedef struct {
+	console_t console;
+	uintptr_t base;
+} console_pl011_t;
+
+int console_pl011_register(console_pl011_t *console, uintptr_t baseaddr,
+			   uint32_t clock, uint32_t baud);
```

Comment:
I think we also need to export console_pl011_init(), or perhaps this can be called console_pl011_early_init()?